client/engine/store/mockstore.go

```diff
@@ -13,35 +14,75 @@ import (
 )
 
 type MockStore struct {
-	objectives map[protocols.ObjectiveId][]byte
-	channels   map[types.Destination][]byte
+	objectives bytesSyncMap
+	channels   bytesSyncMap
 
 	key     []byte        // the signing key of the store's engine
 	address types.Address // the (Ethereum) address associated to the signing key
 }
 
+// bytesSyncMap wraps sync.Map in order to provide type safety
+type bytesSyncMap struct {
+	m sync.Map
+}
+
+// Load returns the value stored in the map for a key, or nil if no
+// value is present.
+// The ok result indicates whether value was found in the map.
+func (o *bytesSyncMap) Load(id string) ([]byte, bool) {
+	obj, ok := o.m.Load(id)
+
+	if !ok {
+		return nil, false
+	}
+
+	return obj.([]byte), ok
+}
+
+// Store sets the value for a key.
+func (o *bytesSyncMap) Store(key string, data []byte) {
+	o.m.Store(key, data)
+}
+
+// Range calls f sequentially for each key and value present in the map.
+// If f returns false, range stops the iteration.
+//
+// Range does not necessarily correspond to any consistent snapshot of the Map's
+// contents: no key will be visited more than once, but if the value for any key
+// is stored or deleted concurrently, Range may reflect any mapping for that key
+// from any point during the Range call.
+//
+// Range may be O(N) with the number of elements in the map even if f returns
+// false after a constant number of calls.
+func (o *bytesSyncMap) Range(f func(key string, value []byte) bool) {
```

Comment:
How do these limitations affect the way we will use `.Range()` (or might use it in future)?